lib/header.ml

```diff
@@ -16,16 +16,30 @@
  *
   }}}*)
 
+exception Found_int of int
+
+let compare_caseless x y =
+  if String.(length x <> length y)
+  then String.compare x y
+  else
+    try
+      for i = 0 to String.length x - 1 do
+        if Char.(lowercase x.[i] <> lowercase y.[i]) then
+          raise (Found_int (Char.compare x.[i] y.[i]))
```

Comment:
> This makes me wonder if we should just use one of the unicode libraries for the header comparison

Certainly not, you are working on the ASCII set, no need to spend time decoding UTF-8.